File: PortalineTrigger.cs
using Celeste.Mod.Entities;
using Microsoft.Xna.Framework;

namespace Celeste.Mod.Portaline {
  [CustomEntity("Portaline/PortalineTrigger")]
  public class PortalineTrigger : Trigger {
    protected bool isEnable;

    public PortalineTrigger(EntityData data, Vector2 offset)
        : base(data, offset) {
      isEnable = data.Bool("is_enable", true);
    }

    public override void OnEnter(Player player) {
      base.OnEnter(player);
      PortalineModule.Session.PortalineEnabled = isEnable;
    }
  }
}

File: Loenn/triggers/PortalineTrigger.lua
local trigger = {}

trigger.name = "Portaline/PortalineTrigger"
trigger.placements = {{
    name = "normal",
    data = {
        width = 16,
        height = 16,
        is_enable = true
    }
}, {
    name = "disable",
    data = {
        width = 16,
        height = 16,
        is_enable = false
    }
}}

return trigger
